esp-metadata/devices/esp32.toml

```diff
@@ -75,3 +75,5 @@ symbols = [
     "pm_support_touch_sensor_wakeup",
     "ulp_supported",
 ]
+
+memory = [{ name = "dram", start = 0x3FFA_E000, end = 0x4000_0000 }]
```

Comment:
we can and should define all memory regions here later and even use them in e.g. the linker scripts

(But that is definitely out of scope for this PR)